tests/Findologic/Api/SortingServiceTest.php

```diff
@@ -70,12 +74,74 @@ public function testProperLegacySortingIsApplied(Request $request, string $expec
         $this->assertSame($expectedOrder, $event->getResult()->getSorting());
     }
 
+    public function availableSortingOptionsProvider(): array
+    {
+        $topsellerSorting = new ProductSortingEntity();
+        $topsellerSorting->setId(Uuid::randomHex());
+        $topsellerSorting->setActive(true);
+        $topsellerSorting->setTranslated([
+            'label' => $this->getContainer()->get('translator')->trans('filter.sortByScore')
+        ]);
+        $topsellerSorting->setKey('product.sales');
+        $topsellerSorting->setPriority(5);
+        $topsellerSorting->setFields([
+            [
+                'field' => 'product.sales',
+                'order' => 'desc',
+                'priority' => 1,
+                'naturalSorting' => 0,
+            ],
+        ]);
+
+        return [
+            'no sorting options' => [
+                'sortings' => new ProductSortingCollection(),
+                'expectedSortings' => [$topsellerSorting]
+            ],
+            'sorting options containing topseller' => [
+                'sortings' => new ProductSortingCollection([$topsellerSorting]),
+                'expectedSortings' => [$topsellerSorting]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider availableSortingOptionsProvider
+     */
+    public function testTopSellerSortingIsOnlyAddedOnce(
+        ProductSortingCollection $sortings,
+        array $expectedSortings
+    ): void {
+        if (Utils::versionLowerThan('6.3.3.0')) {
+            $this->markTestSkipped('Legacy sorting is used for Shopware versions lower than 6.3.3.0');
+        }
+
+        $sortingService = $this->buildSortingService();
+        $event = new ProductListingCriteriaEvent(new Request(), new Criteria(), $this->buildSalesChannelContext());
+        $searchRequestHandlerMock = $this->getMockBuilder(NavigationRequestHandler::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $event->getCriteria()->addExtension('sortings', $sortings);
+        $sortingService->handleRequest($event, $searchRequestHandlerMock);
+
+        $actualSortings = $event->getCriteria()->getExtension('sortings')->getElements();
+        $this->assertCount(count($expectedSortings), $actualSortings);
```

Comment:
Can this be just asserted by count and not by name? E.g. assert(SortingService::TOPSELLER_SORT_FIELD, $actualSorting);

Reply:
Unfortunately we can not, as internally in the plugin we still make use of `_score` sorting, so this would differ. I also didn't wanted to overcomplicate it. We should be fine ensuring that always one sorting is returned if there is none or if there already is a topseller sorting.